apps/shell/app.js

```diff
@@ -72,6 +131,8 @@ wss.on('connection', function connection (ws, req) {
     if (match[2]) dir = decodeURIComponent(match[2]);
   }
 
+  host_in_whitelist = whitelist.includes(host);
```

Comment:
This is now the wrong place for this check. You should both determine the host from the request and validate it is in the whitelist in the `server.on('upgrade', function upgrade(request, socket, head) {` block, which is executed before the `wss.on('connection', function connection (ws, req) {` is executed.